This patch is inspired by upstream commit f9959f03c8241ea1ede5eb4079bec364301aad8d
re-written by Peter Michael Green to apply to zbus 1.0.0

--- a/src/utils.rs	2020-07-16 14:53:32.000000000 +0000
+++ b/src/utils.rs	2021-12-18 21:33:36.446975934 +0000
@@ -39,8 +39,8 @@
                 buf = &buf[n..];
                 cmsg = vec![];
             }
-            Err(nix::Error::Sys(Errno::EINPROGRESS)) => {}
-            Err(nix::Error::Sys(e)) => return Err(e.into()),
+            Err(Errno::EINPROGRESS) => {}
+            Err(e) => return Err(e.into()),
             _ => return Err(Error::new(ErrorKind::Other, "unhandled nix error")),
         }
     }
@@ -79,8 +79,8 @@
                 let tmp = buf;
                 buf = &mut tmp[msg.bytes..];
             }
-            Err(nix::Error::Sys(Errno::EINPROGRESS)) => {}
-            Err(nix::Error::Sys(e)) => return Err(e.into()),
+            Err(Errno::EINPROGRESS) => {}
+            Err(e) => return Err(e.into()),
             _ => return Err(Error::new(ErrorKind::Other, "unhandled nix error")),
         }
     }
